### ISSUE TITLE
Sepa Data Structure (ISO 20022) for swiss ESR v11 File

### ISSUE THREAD

**Issue by metas-mk** [metasfresh/metasfresh#101]
Sepa Data Structure (ISO 20022) for swiss ESR v11 File
- Will become obligatory in 2017

> Check out/ implement the new data structure and provide a new import functionality

**Comment by metasnw** [comment]
https://www.six-interbank-clearing.com/en/home/standardization/iso-payments/customer-bank/implementation-guidelines.html

**Comment by eoss-sh** [comment]
Link to testplatform: https://isotest.postfinance.ch/corporates/?locale=de
Documentation for testplatfrom: https://www.postfinance.ch/de/biz/zv/help/testing.html

**Comment by metas-ts** [comment]
> can we have v11 support and camt 54 in parallel or will there be a hard break?

i think we can support both, but we need to iterate the V11 fields and the CAMT-54 fields one-by-one to see if and where to extend our data structure.

I wonder if CAMT 54 still qualifies as "ESR" (i.e. Einzahlungsschein mit Referenznummer)..depending on the fields comaparison we need to decide if we have the existing widnow more generic, or if we create a new one.

**Comment by eoss-sh** [comment]
This is a camt.54 File one Section of HEV accidentaly got... maybe this is a help as an example:

`<?xml version="1.0" encoding="UTF-8"?>
<Document xmlns="urn:iso:std:iso:20022:tech:xsd:camt.054.001.04" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:iso:std:iso:20022:tech:xsd:camt.054.001.04 camt.054.001.04.xsd"><BkToCstmrDbtCdtNtfctn><GrpHdr><MsgId>20170110375204000370379</MsgId><CreDtTm>2017-01-10T15:50:57</CreDtTm><MsgPgntn><PgNb>1</PgNb><LastPgInd>true</LastPgInd></MsgPgntn><AddtlInf>Reconstruction</AddtlInf></GrpHdr><Ntfctn><Id>20170110375204000370378</Id><CreDtTm>2017-01-10T15:50:57</CreDtTm><FrToDt><FrDtTm>2016-11-11T00:00:00</FrDtTm><ToDtTm>2017-01-09T23:59:59</ToDtTm></FrToDt><CpyDplctInd>DUPL</CpyDplctInd><Acct><Id><IBAN>CH6909000000800420109</IBAN></Id><Ownr><Nm>Hauseigentümerverband Dietikon-Urdorf Dietikon</Nm></Ownr></Acct><Ntry><NtryRef>010466166</NtryRef><Amt Ccy="CHF">68.00</Amt><CdtDbtInd>CRDT</CdtDbtInd><RvslInd>false</RvslInd><Sts>BOOK</Sts><BookgDt><Dt>2016-12-28</Dt></BookgDt><ValDt><Dt>2016-12-29</Dt></ValDt><AcctSvcrRef>20161229007602713965080000000012</AcctSvcrRef><BkTxCd><Domn><Cd>PMNT</Cd><Fmly><Cd>RCDT</Cd><SubFmlyCd>VCOM</SubFmlyCd></Fmly></Domn></BkTxCd><NtryDtls><Btch><NbOfTxs>1</NbOfTxs></Btch><TxDtls><Refs><AcctSvcrRef>161228CH01H0J7EY</AcctSvcrRef><Prtry><Tp>01</Tp><Ref>161228CH01H0J7EY</Ref></Prtry></Refs><Amt Ccy="CHF">68.00</Amt><CdtDbtInd>CRDT</CdtDbtInd><BkTxCd><Domn><Cd>PMNT</Cd><Fmly><Cd>RCDT</Cd><SubFmlyCd>AUTT</SubFmlyCd></Fmly></Domn></BkTxCd><RltdPties><Dbtr><Nm>UBS Switzerland AG</Nm><PstlAdr><StrtNm>Bahnhofstrasse</StrtNm><BldgNb>45</BldgNb><PstCd>8001</PstCd><TwnNm>Zürich</TwnNm><Ctry>CH</Ctry></PstlAdr></Dbtr><DbtrAcct><Id><IBAN>CH1409000000800000022</IBAN></Id></DbtrAcct></RltdPties><RltdAgts><DbtrAgt><FinInstnId><BICFI>POFICHBEXXX</BICFI><Nm>POSTFINANCE AG</Nm><PstlAdr><AdrLine>MINGERSTRASSE 20</AdrLine><AdrLine>3030 BERNE</AdrLine></PstlAdr></FinInstnId></DbtrAgt></RltdAgts><RmtInf><Strd><CdtrRefInf><Tp><CdOrPrtry><Prtry>ISR Reference</Prtry></CdOrPrtry></Tp><Ref>000000003000408208006105897</Ref></CdtrRefInf><AddtlRmtInf>?REJECT?0</AddtlRmtInf></Strd></RmtInf><RltdDts><AccptncDtTm>2016-12-28T20:00:00</AccptncDtTm></RltdDts></TxDtls></NtryDtls><AddtlNtryInf>SAMMELGUTSCHRIFT ESR                    VERARBEITUNG VOM 28.12.2016             KUNDENNUMMER 01-46616-6                 PAKET ID: 161228CH000003FH</AddtlNtryInf></Ntry></Ntfctn></BkToCstmrDbtCdtNtfctn></Document>`

**Comment by metas-rc** [comment]
After talking with Torby and Cristina I understood this is already done. So I am moving it accordingly.
